csharp/src/Microsoft.ML.OnnxRuntime/DisposableNamedOnnxValue.cs

```diff
@@ -101,6 +109,11 @@ private DisposableNamedOnnxValue(string name, Object value, NativeMemoryHandler
             // PinnedMemoryHandle holds the default value as DisposableNamedOnnxValue
             // doesn't hold any managed buffer (that needs to be pinned)
             pinnedMemoryHandle = default;
+            disposeOnnxValueAfterUse = false;
+
+            // set onnx value type and tensor element type
+            onnxValueType = _onnxValueType;
```

Comment:
Now that we have onnxValueType visible to the class, maybe we can make the check on line 101 to just check if the onnxValueType is a plain Tensor ?